### DISCUSSION TITLE
invalid source release error

### DISCUSSION THREAD

**Discussion by mjmay-kinneret** [jplag/JPlag#1851] [Q&A] [answered]
I'm trying run the following command using jplag 5.1.0 on Windows 11

`java -jar jplag-5.1.0-jar-with-dependencies.jar -l java 5784-networks-ass1-*`

I have a series of Java submissions from GitHub classroom in that directory. I'm running the command from the directory with the submissions.  My Java VM version is 22.0.1 (Oracle).  I'm getting the following output errors:

> 2024-07-09-01:52:44_712 [WARN] Submission - Failed to parse submission 5784-networks-ass1-fred\UdpStopWaitServer:
> failed to parse '<null>'error: invalid source release 21 with --enable-preview
>   (preview language features are only supported for release 22)

That's repeated for each submission.  At the end, the tool outputs:

> 2024-07-09-01:52:46_167 [ERROR] CLI - Not enough valid submissions! (found 0 valid submissions)
> 2024-07-09-01:52:46_207 [INFO] SubmissionSet - Summary of all errors:
> 2024-07-09-01:52:44_713 [ERROR] SubmissionSet - ERROR -> Submission 5784-networks-ass1-fred\UdpStopWaitServer removed

I don't get any useful output from the tool.  What is causing the null error?

Thanks.

**Comment by dfuchss** [comment, marked as answer]
We've also encountered this problem in the last week. For us the solution was to use Java JDK 21 instead of Java 22. I hope that helps :)

We'll also fix the behavior in an upcoming release :)
